src/Microsoft.Health.Fhir.CosmosDb.Initialization/Features/Storage/ResourceManagerCollectionSetup.cs

```diff
@@ -0,0 +1,253 @@
+using EnsureThat;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.CosmosDb.Core.Configs;
+using Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage;
+using Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage.StoredProcedures;
+using Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage.Versioning;
+using Polly;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Initialization.Features.Storage;
+
+/// <summary>
+/// Sets up the Cosmos FHIR Database using the ResourceManager SDK and ManagedIdentity.
+/// Requires the MI connection to have 'Cosmos DB Operator' permissions.
+/// </summary>
+public class ResourceManagerCollectionSetup : ICollectionSetup
+{
+    private readonly ILogger<ResourceManagerCollectionSetup> _logger;
+    private readonly ArmClient _armClient;
+    private readonly IOptionsMonitor<CosmosCollectionConfiguration> _cosmosCollectionConfiguration;
+    private readonly CosmosDataStoreConfiguration _cosmosDataStoreConfiguration;
+    private readonly IEnumerable<IStoredProcedureMetadata> _storeProceduresMetadata;
+    private readonly ResourceIdentifier _resourceIdentifier;
+    private CosmosDBSqlDatabaseResource _database;
+    private AzureLocation? _location;
+    private readonly CosmosDBAccountResource _account;
+
+    public ResourceManagerCollectionSetup(
+        IOptionsMonitor<CosmosCollectionConfiguration> cosmosCollectionConfiguration,
+        CosmosDataStoreConfiguration cosmosDataStoreConfiguration,
+        IConfiguration genericConfiguration,
+        IEnumerable<IStoredProcedureMetadata> storedProcedures,
+        ILogger<ResourceManagerCollectionSetup> logger)
+    {
+        EnsureArg.IsNotNull(storedProcedures, nameof(storedProcedures));
+
+        _cosmosCollectionConfiguration = EnsureArg.IsNotNull(cosmosCollectionConfiguration, nameof(cosmosCollectionConfiguration));
+        _cosmosDataStoreConfiguration = EnsureArg.IsNotNull(cosmosDataStoreConfiguration, nameof(cosmosDataStoreConfiguration));
+        _storeProceduresMetadata = EnsureArg.IsNotNull(storedProcedures, nameof(storedProcedures));
+        _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+
+        _armClient = new ArmClient(new DefaultAzureCredential());
+        var dataStoreResourceId = genericConfiguration.GetValue("FhirServer:ResourceManager:DataStoreResourceId", string.Empty);
+        _resourceIdentifier = ResourceIdentifier.Parse(dataStoreResourceId);
+        _account = _armClient.GetCosmosDBAccountResource(_resourceIdentifier);
+    }
+
+    private CosmosDBSqlDatabaseResource Database
+    {
+        get
+        {
+            if (_database != null)
+            {
+                return _database;
+            }
+
+            _database = _account.GetCosmosDBSqlDatabase(_cosmosDataStoreConfiguration.DatabaseId);
+            return _database;
+        }
+    }
+
+    private AzureLocation Location
```

Comment:
[AzureLocation ](https://learn.microsoft.com/en-us/dotnet/api/azure.core.azurelocation?view=azure-dotnet#properties)doesn't support all the regions that we use e.g. EASTUS2EUAP 
if this is just for OSS customers then I think we should be ok

Reply:
This property is used to read the current value from the Cosmos instance since we need to pass it back in to the sdk when updating. We don't set or access the predefined static locations. Is the question if this would fail to read in those regions?

Reply:
Referencing location `new AzureLocation(string)` is also done in common packages and will be in RP code, do you know if there have been issues there?

Reply:
We had a bug in PaaS as AKS creates TM endpoints with entire region name like eastus2euap. We were using the endpoint name to create [AzureLocation earlier ](https://microsofthealth.visualstudio.com/Health/_git/health-paas/pullrequest/34865?_a=files#:~:text=//%20Delete%20an%20endpoint,.DisplayName)))
Which failed for region like eastus2euap which didn't delete the AKS endpoint and tried to create SF endpoint with cluster short id as eusep (During AKS to SF migration). Assigning the same priority to both failed 

I was just wondering if we will be missing out on any regions ?